org.eclipse.cdt.cross.arc.gnu/src/org/eclipse/cdt/cross/arc/gnu/common/ResourceChangeListener.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * This program and the accompanying materials are made available under the terms of the Common
+ * Public License v1.0 which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/cpl-v10.html
+ *
+ * Copyright (c) 2016 Synopsys, Inc.
+ *******************************************************************************/
+
+package org.eclipse.cdt.cross.arc.gnu.common;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IResourceChangeEvent;
+import org.eclipse.core.resources.IResourceChangeListener;
+import org.eclipse.core.resources.IResourceDelta;
+
+/**
+ * This class is intended for tracking the resource's changes.
+ */
+public class ResourceChangeListener implements IResourceChangeListener {
+
+  private static ResourceChangeListener INSTANCE = null;
+
+  public synchronized static ResourceChangeListener getInstance(
+      final ProjectEventListener listener) {
+    if (INSTANCE == null) {
+      INSTANCE = new ResourceChangeListener(listener);
+    }
+    return INSTANCE;
+  }
+
+  private ProjectEventListener listener;
+
+  public interface ProjectEventListener {
```

Comment:
Interfaces names should start with `I`. Please include some commentary before class and interface definitions that would answer the question "Why this class/interface is needed?".